Tbabula

tbabula-slac opened the pull request:
**Description of PR**
<!--- Describe what the PR holds -->
changes to ansible-role-telegraf template

**Type of change**
adding properties for submodules and fields to support generating swh.conf file

Feature Pull Request
Bugfix Pull Request
Docs Pull Request

**Fixes an issue**
original playbook seemed too limited to generate proper config file for this specific use case:

Here is what I found what worked/did not work for me:
By default configuration it worked for  generating 10-power but not 20-swh without template modification.
the closest way to configure sumbodule as it is was to define filter.name with parameters with filter.config but writing another filter.name was overriding the previous and only creates [] instead of [[]] in the header
One plugin per file works, another plugin overrides previous and get warning about duplicate in dictionary so even using something like plugin: name.submodule-name wouldn't work . Input name is based on dict key which can't be duplicated.
inputs.{{ item.key }}]

Commit:
extending telegraf

Commit:
merging changes into telegraf_extra_plugins environment

Commit:
Needs {% if item.value.subfields is defined and item.value.subfields is iterable %} so it doesn not fail when subfields is not definied

Commit:
fixing syntax for if conditional, missing {% endif %}

yee379 commented:
is `telegraf-custom-extra` really necessary? at first glance it doesn't appear that your additions to `telegraf_extra_plugins` should break anything for us.

tbabula-slac commented:
there should be only telegraf-extra-plugins left as I deleted the other one from configure_linux.yml task, custom duplicate template and renamed reference in inventory back to telegraf-extra-plugins.
